### PR TITLE
Cached api controller pipelines

### PR DESCRIPTION
**Pull request by allouis**
ref ENG-761
ref https://linear.app/tryghost/issue/ENG-761

Creating these pipelines is expensive, and we don't want to do it repeatedly for the same controller. Adding caching should reduce the amount of time spent setting up pipelines for each usage of the `get` helper.

### CODE REVIEW

File: ghost/api-framework/lib/pipeline.js [hunk 1]
```diff
@@ -267,6 +273,10 @@ const pipeline = (apiController, apiUtils, apiType) => {
         Object.assign(obj[method], apiImpl);
         return obj;
     }, {});
+
+    controllerMap.set(apiController, result);
```
Comment (daniellockyer) [inline ghost/api-framework/lib/pipeline.js:277]: Would it be worth using `docName` as the key?

Comment (allouis) [reply ghost/api-framework/lib/pipeline.js:277]: `docName` isn't unique - so no! 😝

Comment (daniellockyer) [reply ghost/api-framework/lib/pipeline.js:277]: wtf

Comment (allouis) [reply ghost/api-framework/lib/pipeline.js:277]: One of the things `docName` is used for is to determine which permissions to use - so the `pages` endpoint has a `docName` of `'posts'` - as does the `postsPublic` endpoint

Comment (daniellockyer) [reply ghost/api-framework/lib/pipeline.js:277]: unbelievable

**Pull request merged**